src/java.base/share/classes/sun/net/www/http/KeepAliveCache.java

```diff
@@ -122,6 +122,7 @@ public KeepAliveCache() {}
      */
     @SuppressWarnings("removal")
     public void put(final URL url, Object obj, HttpClient http) {
+        HttpClient oldClient = null;
```

Comment:
Could you put a comment in there explaining what oldClient is exactly? Eg that it is a client that ends up being closed, either from the cache to make way for the newest entry or if the current client needs to be closed.

Reply:
Added. Let me know if that's what you had in mind.